rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationSupportHandler.java

```diff
@@ -46,4 +63,86 @@ public IntegrationSupportHandler( ProjectGenerator projectConverter) {
         return projectConverter.generatePom(integration);
     }
 
+
+    @POST
+    @Path("/import")
+    public Response importIntegration(InputStream is) {
+        try (ZipInputStream zis = new ZipInputStream(is)) {
+            int imported = 0;
+            while (true) {
+                ZipEntry entry = zis.getNextEntry();
+                if( entry == null ) {
+                    break;
+                }
+                if (EXPORT_MODEL_FILE_NAME.equals(entry.getName())) {
+                    ModelData[] models = Json.mapper().readValue(new FilterInputStream(zis) {
+                        @Override
+                        public void close() throws IOException {
+                            // We want to avoid closing zis
+                        }
+                    }, ModelData[].class);
+
+                    imported += importModels(models);
+                }
+                zis.closeEntry();
+            }
+            if (imported==0) {
+                LOG.info("Could not import integration: No integration data model found.");
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+            return Response.status(Response.Status.NO_CONTENT).build();
+        } catch (IOException e) {
+            if (LOG.isInfoEnabled()) {
+                LOG.info("Could not import integration: " + e, e);
+            }
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+    }
+
+    public int importModels(ModelData... models) throws IOException {
+        int count = 0;
+        for (ModelData model : models) {
+            switch (model.getKind()) {
```

Comment:
Could this be simpler? We can get `modelClass` from `Kind` and could have a constant `EnumSet<Kind> ALLOWED_IMPORTS`?

Reply:
Aah I see Connections/Connectors are handled differently

Reply:
Right!